builder/docker/communicator.go

```diff
@@ -194,8 +194,34 @@ func (c *Communicator) UploadDir(dst string, src string, exclude []string) error
 	return nil
 }
 
+// Download pulls a file out of a container using `docker cp`. We have a source
```

Comment:
Upon re-reading the docs for the [cp](https://docs.docker.com/reference/commandline/cp/) command, it looks like this could be anything that exists on the filesystem. The important distinction is file vs. directory, since that has special treatment for the upload section. I've only tested with files so far though. Not sure if you mean "binary" to mean something else.